le_dash/banner.py

```diff
@@ -1,18 +1,37 @@
 import six
+import hashlib
 import requests
 from django.conf import settings
+from django.core.cache import cache
+import logging
 
 if six.PY2:
     from urlparse import urljoin
 else:
     from urllib.parse import urljoin
 
 
+logger = logging.getLogger(__name__)
+
+
 def banner_req(url, params):
+    #  md5 is cheaper than urlunparse/urlencode params
+    get_url = hashlib.md5("{0}".format(params))
```

Comment:
I could use it. I cached the full url string the first rev and added urlencode, urlunparse etc to build the url, but since the url is the same in all cases and the params are different. I opted to use the md5(params) instead. I used debug logs and it worked fine. I got some time savings, but d3 is still a bit slow, so I am doing some more optimizations.